---
title: "DATA ANALYSIS 1 - Dynamics of the human nasal microbiota and Staphylococcus aureus CC398 carriage in pig truck drivers across one workweek"
output: 
  html_document: 
    toc: yes
    toc_depth: 2
    df_print: kable
author: Anna Ingham, Statens Serum Institut, Denmark
date: "February 2021"
editor_options: 
  chunk_output_type: inline
---

### Preparation steps

##### Load packages
```{r include=FALSE}
library(phyloseq); packageVersion("phyloseq")
library(ggplot2); packageVersion("ggplot2")
library(dplyr); packageVersion("dplyr")
library(tibble); packageVersion("tibble")
library(Biostrings); packageVersion("Biostrings")
library(vegan); packageVersion("vegan")
library(openxlsx)
library(tidyr)
library(ggrepel)
library(genefilter)
library(PMCMR)
library(DescTools)
library(lme4)
library(lmerTest)
library(ggResidpanel)
library(ComplexHeatmap)
library(genefilter)
library(cluster)
library(factoextra)
library(RColorBrewer)
library(markovchain)
library(diagram)
#devtools::install_github("easystats/report")
library(report)
library(modelbased)
library(ranacapa)
library(splinectomeR)
library(sm)
library(ggalluvial)
library(labdsv)
library(khroma)
library(reshape2)
theme_set(theme_bw())
```



##### Read phyloseq object 
```{r echo=TRUE}
ps1 <- readRDS("./R_DATA/ps1_pig_truck_drivers.RData")
ps1
```

##### Sequencing depth
```{r echo=TRUE, fig.height=4, fig.width=10}
sdt = data.table::data.table(as(sample_data(ps1), "data.frame"),
                 TotalReads = sample_sums(ps1), keep.rownames = TRUE)
data.table::setnames(sdt, "rn", "SampleID")
pSeqDepth = ggplot(sdt, aes(TotalReads)) + geom_histogram(binwidth = 1000) + ggtitle("Sequencing depth") + theme(plot.title = element_text(size = 14, face = "bold"), axis.text.x=element_text(angle = 45, vjust=1, hjust=1)) + labs(y="# of samples", x ="Post-QC library size per sample") + scale_x_continuous(breaks = scales::pretty_breaks(n = 25))
pSeqDepth + facet_grid(Subject_type~., space="free", scales ="free") 
```


## **Compare alpha diversity in the nose by the Shannon index**
#### Other diversity indices were also explored and results were similar for all indices. Shannon diversity was chosen for the final analyses. 
```{r echo=TRUE}
#### Add diversity measures to the phyloseq object as variables
alpha_div_raw <- estimate_richness(ps1, measures=c("Observed", "Chao1", "Shannon", "InvSimpson")) 
ps1 <- merge_phyloseq(ps1, sample_data(alpha_div_raw))
df_ps1 <- as(sample_data(ps1), "data.frame")
```


##### Before comparing each daytime of the drivers against spouses and non-exposed, first test whether diversity differs by weekday
##### Linear models
##### Morning:
```{r}
fit2 <- lmer(Shannon ~ Day + (1|Subject_ID), data = df_ps1[df_ps1$Subject_type=="driver" & df_ps1$Time_point_within_day_long == "A_morning",])
summary(fit2)
```


#### Test overall model significance (with anova)
```{r }
anova(fit2)
```

#### The overall model is not significant.

### Check model diagnostics to see if the assumptions are met and if we can trust the results
```{r  fig.width=10}
resid_panel(fit2, plots = "all", qqbands = TRUE)
```



#### Do a Post-hoc test / Contrast Analysis (This changes the reference level in the model)
```{r }
results2 <-estimate_contrasts(fit2, "Day", adjust = "fdr")
print(results2)
```

Morning samples from all weekdays are similar, also when accounting for Subject_ID. 


##### After first unloading:
```{r}
fit3 <- lmer(Shannon ~ Day + (1|Subject_ID), data = df_ps1[df_ps1$Subject_type=="driver" & df_ps1$Time_point_within_day_long == "B_after_first_unloading",])
summary(fit3)
```


#### Test overall model significance (with anova)
```{r}
anova(fit3)
```

#### The overall model is not significant.

### Check model diagnostics to see if the assumptions are met and if we can trust the results
```{r  fig.width=10}
resid_panel(fit3, plots = "all", qqbands = TRUE)
```
 


#### Do a Post-hoc test / Contrast Analysis (This changes the reference level in the model)
```{r  }
results2 <-estimate_contrasts(fit3, "Day", adjust = "fdr")
print(results2)
```

First_unloading samples from all weekdays are similar, also when accounting for Subject_ID. There is a trend for a lower diversity after first unloading on Thursday and Friday compared with Monday. It is not significant in the contrast analysis, but is significant in the model, likely because the model does not apply multiple testing correction while the contrast analysis does. 

##### After second unloading:
```{r}
fit4 <- lmer(Shannon ~ Day + (1|Subject_ID), data = df_ps1[df_ps1$Subject_type=="driver" & df_ps1$Time_point_within_day_long == "C_after_second_unloading",])
summary(fit4)
```


#### Test overall model significance (with anova)
```{r }
anova(fit4)
```

#### The overall model is not significant.

### Check model diagnostics to see if the assumptions are met and if we can trust the results
```{r fig.width=10}
resid_panel(fit4, plots = "all", qqbands = TRUE)
```


#### Do a Post-hoc test / Contrast Analysis (This changes the reference level in the model)
```{r  }
results2 <-estimate_contrasts(fit4, "Day", adjust = "fdr")
print(results2)
```

Second_unloading samples from all weekdays are similar, also when accounting for Subject_ID.  



Since we verified that drivers' diversity does not significantly differ by weekday, we can now compare the daytimes (pooled by weekday) to spouses and non-exposed.    



### Shannon diversity over time (within a day): 
```{r echo=TRUE, fig.height=5, fig.width=5}
plot_g_Shannon <- ggplot(df_ps1, aes(x = Time_point_within_day_long, y = Shannon, fill = Subject_type, color=Subject_type)) + geom_boxplot(outlier.color="NA", alpha = 0.5) +  geom_dotplot(binaxis = "y", stackdir = "center", alpha = 0.8, position=position_dodge(0.75), dotsize = 0.5) + theme(axis.title.y = element_text(size=12, face = "bold"), axis.text.y = element_text(size=16), axis.text.x = element_text(size=14, face = "bold", angle = 0), axis.title.x = element_blank(), legend.text = element_text(size=10, face = "bold"), legend.title = element_blank(), panel.grid.major = element_blank(), panel.background = element_blank(), panel.border = element_blank(), axis.line = element_line(colour = "black"), title = element_text(size = 14, face = "bold"), legend.position = "bottom") + labs(y="Shannon Diversity") + stat_boxplot(geom ='errorbar')   + scale_fill_manual(values = c("#00AFBB","#FF9505","#FC4E07")) + scale_color_manual(values = c("#00AFBB","#FF9505","#FC4E07")) + scale_x_discrete(labels=c("A_morning" = "Morning", "B_after_first_unloading" = "1", "C_after_second_unloading" = "2", "non_exposed" = "non-exposed"))

plot_g_Shannon
```


### Test if diversity differs between drivers and spouses in the mornings (and compared to non_exposed subjects)
```{r}
df_ps1_morn = df_ps1[df_ps1$Time_point_within_day_long %in% c("A_morning","non_exposed"),]
pairwise.wilcox.test(df_ps1_morn$Shannon,df_ps1_morn$Subject_type, paired = FALSE, p.adjust.method = "BH")
```

Non-exposed individuals have significantly higher diversity than drivers and their spouses in the mornings.There is a trend for higher diversity in spouses compared to drivers, but it is not significant. 

### Test if diversity differs between drivers after 1st unloading and non_exposed subjects 
```{r}
df_ps1_1st = df_ps1[df_ps1$Time_point_within_day_long %in% c("B_after_first_unloading","non_exposed"),]
pairwise.wilcox.test(df_ps1_1st$Shannon,df_ps1_1st$Subject_type, paired = FALSE, p.adjust.method = "BH")
```

Non-exposed individuals have significantly higher diversity than drivers after 1st unloading.



### Test if diversity differs between drivers after 2st unloading and non_exposed subjects 
```{r}
df_ps1_2nd = df_ps1[df_ps1$Time_point_within_day_long %in% c("C_after_second_unloading","non_exposed"),]
pairwise.wilcox.test(df_ps1_2nd$Shannon,df_ps1_2nd$Subject_type, paired = FALSE, p.adjust.method = "BH")
```

Non-exposed individuals do not differ in diversity from drivers after 2nd unloading.



### Test if difference between daytimes is significant (drivers only) and take into account repeated measuremnts by subject and days during the week (include as random effects).

#### Fit linear model
```{r echo=TRUE}
fit <- lmer(Shannon ~ Time_point_within_day_long + (1|Subject_ID) + (1|Day), data = df_ps1[df_ps1$Subject_type=="driver",])
summary(fit)
```

#### Test overall model significance (with anova)
```{r echo=TRUE}
anova(fit)
```

#### The overall model is significant, suggesting that Shannon differs between daytimes

### Check model diagnostics to see if the assumptions are met and if we can trust the results
```{r echo=TRUE, fig.width=10}
resid_panel(fit, plots = "all", qqbands = TRUE)
```

#### Based on the quantile-quantile plot, normality of the residuals can be assumed (supported also by the histogram)
#### Residuals vs observed (index plots) shows the desired random distribution
#### Therefore we can trust the lme results


#### To see which time points are different from each other, do a Post-hoc test / Contrast Analysis (This changes the reference level in the model)
```{r echo=TRUE}
results <-estimate_contrasts(fit, "Time_point_within_day_long", adjust = "fdr")
print(results)
```



Bacterial alpha diversity increases significantly from morning to first unloading. Diversity does not signicantly increase further to second unloading.    


#### Plot the Shannon diversity of drivers and spouses, visualizing similar patterns within each weekday 
```{r, fig.height=5, fig.width=8}
df_ps1$Day_int <- as.numeric(as.character(df_ps1$Day_int))

ggplot(df_ps1[df_ps1$Subject_type != "non_exposed",], aes(x = Consecutive_time_point, y = Shannon))  + geom_rect(aes(NULL, NULL, xmin = Day, xmax = dplyr::lag(Day), fill = Day),ymin = -0.5, ymax = Inf, data = df_ps1[df_ps1$Subject_type != "non_exposed",], show.legend = FALSE) + scale_fill_manual(values = c("white", "grey", "white", "grey", "white", "grey")) + geom_boxplot(aes(x = Consecutive_time_point, y = Shannon)) + stat_boxplot(geom ='errorbar') + geom_dotplot(binaxis = "y", stackdir = "center", alpha=0.75, dotsize=0.75) + stat_summary(fun=median, colour="red", geom="line", aes(group=1), size =1) + theme(axis.title.y = element_text(size=12, face = "bold"), axis.text.y = element_text(size=16), axis.text.x = element_text(size=10, face = "bold", angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.text = element_text(size=14, face = "bold"), legend.title = element_text(face = "bold", size = 20), panel.grid.major = element_blank(), panel.border = element_blank(), axis.line = element_line(colour = "black"), title = element_text(size = 14, face = "bold"), strip.text.x = element_text(size = 14, face = "bold"), strip.background = element_rect(fill = "white")) + labs(y="Shannon Diversity")  + facet_wrap(.~Subject_type, nrow=2) + scale_x_discrete(labels=c("1_Monday" = "", "1_Monday_B_after_first_unloading" = "Monday 1", "1_Monday_C_after_second_unloading" = "Monday 2", "2_Tuesday" = "","2_Tuesday_A_morning" = "Tuesday morning", "2_Tuesday_B_after_first_unloading" = "Tuesday 1", "2_Tuesday_C_after_second_unloading" = "Tuesday 2", "3_Wednesday" = "","3_Wednesday_A_morning" = "Wednesday morning", "3_Wednesday_B_after_first_unloading" = "Wednesday 1", "3_Wednesday_C_after_second_unloading" = "Wednesday 2","4_Thursday" = "","4_Thursday_A_morning" = "Thursday morning", "4_Thursday_B_after_first_unloading" = "Thursday 1", "4_Thursday_C_after_second_unloading" = "Thursday 2","5_Friday" = "","5_Friday_A_morning" = "Friday morning", "5_Friday_B_after_first_unloading" = "Friday 1", "5_Friday_C_after_second_unloading" = "Friday 2",  "8_Monday" = "","8_Monday_A_morning" = "Monday morning"))
```






## **Ordination analyses**

#### Hellinger transform the count data
```{r}
ps1_hell <- transform_sample_counts(ps1, function(x) sqrt(x / sum(x))) #hellinger transform
summary(sample_sums(ps1_hell))

sample_data(ps1_hell)$Subject_type_Daytime <- paste(sample_data(ps1_hell)$Subject_type, sample_data(ps1_hell)$Time_point_within_day_long, sep = "_")
```


### PCoA on Bray-Curtis distance matrix
```{r}
ps1_hell_ord <- ordinate(ps1_hell, method = "PCoA", distance = "bray")
```


### PCoA - Axis 1,2
```{r}
ord_plot <- plot_ordination(ps1_hell, ps1_hell_ord, type="samples", color="Subject_type_Daytime", axes = 1:2)
ord_plot1 <- ord_plot + stat_ellipse(geom = "polygon", type="norm", alpha=0.3, aes(fill=Subject_type_Daytime)) + scale_fill_manual(values = c("#b2e7ea","#008c95","#002325","#FF9505","#FC4E07"), name = "", labels = c("Driver - Morning", "Driver - after 1. unloading", "Driver - after 2. unloading", "Non-exposed", "Spouse")) + scale_color_manual(values = c("#b2e7ea","#008c95","#002325","#FF9505","#FC4E07")) + guides(color = FALSE, fill = guide_legend(override.aes = list(alpha = 1))) + geom_point()

ord_plot1
```


### PCoA - Axis 1,3
```{r}
ord_plot <- plot_ordination(ps1_hell, ps1_hell_ord, type="samples", color="Subject_type_Daytime", axes = c(1,3))
ord_plot + stat_ellipse(geom = "polygon", type="norm", alpha=0.3, aes(fill=Subject_type_Daytime)) + scale_fill_manual(values = c("#b2e7ea","#008c95","#002325","#FF9505","#FC4E07"), name = "", labels = c("Driver - Morning", "Driver - after 1. unloading", "Driver - after 2. unloading", "Non-exposed", "Spouse")) + scale_color_manual(values = c("#b2e7ea","#008c95","#002325","#FF9505","#FC4E07")) + guides(color = FALSE, fill = guide_legend(override.aes = list(alpha = 1))) + geom_point()
```



### PCoA - Axis 2,3
```{r}
ord_plot <- plot_ordination(ps1_hell, ps1_hell_ord, type="samples", color="Subject_type_Daytime", axes = c(2,3))
ord_plot + stat_ellipse(geom = "polygon", type="norm", alpha=0.3, aes(fill=Subject_type_Daytime)) + scale_fill_manual(values = c("#b2e7ea","#008c95","#002325","#FF9505","#FC4E07"), name = "", labels = c("Driver - Morning", "Driver - after 1. unloading", "Driver - after 2. unloading", "Non-exposed", "Spouse")) + scale_color_manual(values = c("#b2e7ea","#008c95","#002325","#FF9505","#FC4E07")) + guides(color = FALSE, fill = guide_legend(override.aes = list(alpha = 1))) + geom_point()
```



#### **Analysis of similarities (anosim())**

#### Is the bacterial community different between morning and after 1st unloading? (taking into account subject-wise repeated measurements) (drivers only)
```{r fig.height=3, fig.width=7}
ps1_hell_drivers <- subset_samples(ps1_hell, Subject_type=="driver" & Time_point_within_day_long %in% c("A_morning", "B_after_first_unloading"))
  
df <- as(sample_data(ps1_hell_drivers), "data.frame")

bray_dist <- phyloseq::distance(ps1_hell_drivers, method = "bray")

set.seed(123)
dt_ano <- with(df, anosim(bray_dist,Time_point_within_day_long, strata=df$Subject_ID))
summary(dt_ano)
plot(dt_ano)
```


P<0.05, therefore the bacterial community in the drivers' noses is different between morning and after 1st unloading.  
Morning samples have a higher within-group variation. This is also visible from the PCoA. However, between group variation is still highest. 


#### Is the bacterial community different between morning and after 2nd unloading? (taking into account subject-wise repeated measurements) (drivers only)
```{r fig.height=3, fig.width=7}
ps1_hell_drivers <- subset_samples(ps1_hell, Subject_type=="driver" & Time_point_within_day_long %in% c("A_morning", "C_after_second_unloading"))
  
df <- as(sample_data(ps1_hell_drivers), "data.frame")

bray_dist <- phyloseq::distance(ps1_hell_drivers, method = "bray")

set.seed(123)
dt_ano <- with(df, anosim(bray_dist,Time_point_within_day_long, strata=df$Subject_ID))
summary(dt_ano)
plot(dt_ano)
```

P<0.05, therefore the bacterial community in the drivers' noses is different between morning and after 2nd unloading.  


#### Is the bacterial community different between after 1st and after 2nd unloading? (taking into account subject-wise repeated measurements) (drivers only)
```{r fig.height=3, fig.width=7}
ps1_hell_drivers <- subset_samples(ps1_hell, Subject_type=="driver" & Time_point_within_day_long %in% c("B_after_first_unloading", "C_after_second_unloading"))
  
df <- as(sample_data(ps1_hell_drivers), "data.frame")

bray_dist <- phyloseq::distance(ps1_hell_drivers, method = "bray")

set.seed(123)
dt_ano <- with(df, anosim(bray_dist,Time_point_within_day_long, strata=df$Subject_ID))
summary(dt_ano)
plot(dt_ano)
```


P<0.05, therefore the bacterial community in the drivers' noses is different between 1st and 2nd unloading.  
However, the within-group variation after 2nd unloading is higher than the between group variation.  




### Is the community structure of drivers and spouses different in the morning? 
```{r fig.height=3, fig.width=7}
ps1_hell_morning <- subset_samples(ps1_hell, Time_point_within_day_long=="A_morning" & Subject_type !="non_exposed")
  
df <- as(sample_data(ps1_hell_morning), "data.frame")

bray_dist <- phyloseq::distance(ps1_hell_morning, method = "bray")

set.seed(123)
s_ano <- with(df, anosim(bray_dist,Subject_type))
summary(s_ano)
plot(s_ano)
```


P<0.05, therefore the bacterial community of drivers and spouses in the morning is different. 




### Are the non_exposed samples different from spouses' morning samples?
```{r fig.height=3, fig.width=7}
ps1_hell_morningS <- subset_samples(ps1_hell, Subject_type %in% c("spouse","non_exposed"))
  
df <- as(sample_data(ps1_hell_morningS), "data.frame")

bray_dist <- phyloseq::distance(ps1_hell_morningS, method = "bray")

set.seed(123)
mornS_ano <- with(df, anosim(bray_dist,Subject_type))
summary(mornS_ano)
plot(mornS_ano)
```

P<0.05, therefore the bacterial community of non-exposed and spouses is different. 



### Are the non_exposed samples different from drivers' morning samples?
```{r  fig.height=3, fig.width=7}
ps1_hell_morning <- subset_samples(ps1_hell, Time_point_within_day_long %in% c("A_morning","non_exposed") & Subject_type %in% c("driver","non_exposed"))
  
df <- as(sample_data(ps1_hell_morning), "data.frame")

bray_dist <- phyloseq::distance(ps1_hell_morning, method = "bray")

set.seed(123)
morn_ano <- with(df, anosim(bray_dist,Subject_type))
summary(morn_ano)
plot(morn_ano)
```

P<0.05, therefore the bacterial community of non-exposed and drivers in the morning is different.  


### Are the non_exposed samples different from drivers' 1st unloading samples?
```{r  fig.height=3, fig.width=7}
ps1_hell_1st <- subset_samples(ps1_hell, Time_point_within_day_long %in% c("B_after_first_unloading","non_exposed") & Subject_type %in% c("driver","non_exposed"))
  
df <- as(sample_data(ps1_hell_1st), "data.frame")

bray_dist <- phyloseq::distance(ps1_hell_1st, method = "bray")

set.seed(123)
B_ano <- with(df, anosim(bray_dist,Subject_type))
summary(B_ano)
plot(B_ano)
```

P<0.05, therefore the bacterial community of non-exposed and drivers after 1st unloading is different.  




### Are the non_exposed samples different from drivers' 2nd unloading samples?
```{r fig.height=3, fig.width=7}
ps1_hell_2nd <- subset_samples(ps1_hell, Time_point_within_day_long %in% c("C_after_second_unloading","non_exposed") & Subject_type %in% c("driver","non_exposed"))
  
df <- as(sample_data(ps1_hell_2nd), "data.frame")

bray_dist <- phyloseq::distance(ps1_hell_2nd, method = "bray")

set.seed(123)
C_ano <- with(df, anosim(bray_dist,Subject_type))
summary(C_ano)
plot(C_ano)
```

P<0.05, therefore the bacterial community of non-exposed and drivers after 2nd unloading is different.  





## **Bacterial load (qPCR)** 

### Number of samples above standard curve threshold
```{r}
table(df_ps1$Subject_type, df_ps1$Quantity>1000000)
```


### Bacterial load per sample in drivers and spouses
```{r, fig.height=5, fig.width=5}
plot_load1 <- ggplot(df_ps1[! df_ps1$Subject_type %in% "non_exposed",], aes(x = Time_point_within_day_long, y = Quantity_per_swab, fill = Subject_type, color=Subject_type)) + geom_boxplot(outlier.color="NA", alpha = 0.5) +  geom_dotplot(binaxis = "y", stackdir = "center", alpha = 0.8, position=position_dodge(0.75), dotsize = 0.5) + theme(axis.title.y = element_text(size=12, face = "bold"), axis.text.y = element_text(size=16), axis.text.x = element_text(size=12, face = "bold", angle = 0), axis.title.x = element_blank(), legend.text = element_text(size=10, face = "bold"), legend.title = element_blank(), panel.grid.major = element_blank(), panel.background = element_blank(), panel.border = element_blank(), axis.line = element_line(colour = "black"), title = element_text(size = 14, face = "bold"), legend.position = "bottom") + labs(y="16S rRNA gene copies per sample") + stat_boxplot(geom ='errorbar')   + scale_fill_manual(values = c("#00AFBB","#FC4E07")) + scale_color_manual(values = c("#00AFBB","#FC4E07")) + scale_y_log10() + scale_x_discrete(labels=c("A_morning" = "Morning", "B_after_first_unloading" = "1", "C_after_second_unloading" = "2"))

plot_load1
```


It was decided not to perform statistical testing on the qPCR data because most driver and spouse samples exceeded the highest value of the standard curve.  






## **Community state typing (CSTs)**

### Clustering into CSTs by partitioning around medoid (PAM) clustering
### Perform clustering across all samples and time points based on Jenson-Shannon distance


#### Decide on the number of dimensions: Determine where the decrease of eigenvalues levels off.
```{r fig.height=5, fig.width=10, message=FALSE}
#distance matrix:
jsd_dist <- phyloseq::distance(ps1_hell, method = "jsd")

ord = ordinate(ps1_hell, method = "MDS", distance = jsd_dist)
plot_scree(ord) + ggtitle("MDS-jsd ordination eigenvalues") + theme(axis.title.x = element_text(size = 20), axis.title.y = element_text(size = 20), axis.text.x = element_text(size = 11), axis.text.y = element_text(size = 13)) + xlim(as.character(seq(1,20)))
```


```{r}
evs <- ord$value$Eigenvalues
print(evs[1:20])

print(tail(evs))
```


The scree plot suggests that the decrease levels off from around 5/7. 

### Determine the number of clusters

#### To identify the most appropriate number of clusters to separate the data into, we employ the elbow method, silhouette method, and gap statistic with 5 MDS dimensions.

```{r , fig.height=14}
NDIM <- 5
x <- ord$vectors[,1:NDIM]

pamPCoA = function(x, k) {
    list(cluster = pam(x[,1:2], k, cluster.only = TRUE))
}
gs = clusGap(x, FUN = pamPCoA, K.max = 12, B = 500)
```


```{r}
# Elbow method
fviz_nbclust(x=x, pamPCoA, method = "wss") +
  labs(subtitle = "Elbow method") 

# Silhouette method
fviz_nbclust(x=x, pamPCoA, method = "silhouette") +
  labs(subtitle = "Silhouette method")

# Gap statistic
set.seed(123)
fviz_gap_stat(gs, maxSE = list(method = "Tibs2001SEmax")) +
  labs(subtitle = "Gap statistic method") 
```

#### Elbow method: Local minimum: 3
#### Highest silhouette width at: 3
#### Gap statistic: Local maximum: 3

#### Let's go with 3 clusters as suggested.


### **Cluster into CSTs**

#### Perform partitioning around medoids (PAM) clustering:
```{r}
K <- 3
x <- ord$vectors[,1:NDIM]
data.cluster <- pam(x, k=K, cluster.only=T)
clust <- as.factor(pam(x, k=K, cluster.only=T))
sample_data(ps1_hell)$CST <- clust
sample_data(ps1)$CST <- clust
CSTs <- as.character(seq(K))
```


### **Evaluate clustering**

#### Inspect the results in MDS ordinations:
```{r, fig.width=6}
CSTColors <- brewer.pal(3,"Dark2")
CSTColors <- CSTColors[c(3,2,1)]
names(CSTColors) <- CSTs
CSTColorScale <- scale_colour_manual(name = "CST", values = CSTColors[1:3])
CSTFillScale <- scale_fill_manual(name = "CST", values = CSTColors[1:3])

plot_ordination(ps1_hell, ord, color="CST")+ CSTColorScale + geom_point(size=5)  + theme(axis.title.x = element_text(size = 20), axis.title.y = element_text(size = 20), axis.text.x = element_text(size = 13), axis.text.y = element_text(size = 13))#axes 1 and 2 

plot_ordination(ps1_hell, ord, axes=c(1,3), color="CST") + CSTColorScale + geom_point(size=5) + theme(axis.title.x = element_text(size = 20), axis.title.y = element_text(size = 20), axis.text.x = element_text(size = 13), axis.text.y = element_text(size = 13))#axes 1 and 3

plot_ordination(ps1_hell, ord, axes=c(2,3), color="CST") +CSTColorScale + geom_point(size=5) + theme(axis.title.x = element_text(size = 20), axis.title.y = element_text(size = 20), axis.text.x = element_text(size = 13), axis.text.y = element_text(size = 13))#axes 2 and 3
```





### Indicator taxa analysis for each CST (on raw counts, ASV level)
```{r}
df_CST <- as(sample_data(ps1), "data.frame")
df_CST <- df_CST %>% select(CST)

ps1 <- prune_taxa(taxa_sums(ps1) != 0, ps1)

m_ps1_ot <- as(otu_table(ps1)@.Data, "matrix")
df_ps1_ot <- as.data.frame(m_ps1_ot)
```


```{r}
set.seed(123)
indic <- indval(df_ps1_ot, clustering=df_CST$CST, numitr=500)
```

Indicator values (IV) range from 0 to 1, with higher values for stronger indicators. 

Indicators per cluster (both non-significant and significant)
```{r}
table(indic$maxcls)
```

#### Significant indicators per CST before FDR correction
```{r}
summary(indic, p=0.05, type='short', digits=2, show=p,sort=FALSE)
```

#### Correct for multiple testing 
```{r}
indic$pval_original <- indic$pval
indic$pval <- p.adjust(indic$pval, "fdr")
```

#### Significant indicators per CST after FDR correction
```{r}
summary(indic, p=0.05, type='short', digits=2, show=p,sort=FALSE)
```
No significant indicators left after multiple testing correction on ASV level.  

#### On genus level:
### Indicator taxa analysis for each CST (on raw counts, genus level)
```{r}
ps1_genus <- tax_glom(ps1, taxrank="Genus")

m_ps1_ot_G <- as(otu_table(ps1_genus)@.Data, "matrix")
df_ps1_ot_G <- as.data.frame(m_ps1_ot_G)
```


```{r}
set.seed(123)
indic_G <- indval(df_ps1_ot_G, clustering=df_CST$CST, numitr=500)
```

Indicator values (IV) range from 0 to 1, with higher values for stronger indicators. 

Indicators per cluster (both non-significant and significant)
```{r}
table(indic_G$maxcls)
```

#### Significant indicators per CST before FDR correction
```{r}
summary(indic_G, p=0.05, type='short', digits=2, show=p,sort=FALSE)
```

#### Correct for multiple testing 
```{r}
indic_G$pval_original <- indic_G$pval
indic_G$pval <- p.adjust(indic_G$pval, "fdr")
```

#### Significant indicators per CST after FDR correction
```{r}
summary(indic_G, p=0.05, type='short', digits=2, show=p,sort=FALSE)
```

On genus level there are 105 significant indicator genera left after multiple testing correction.  
CST one has the most (69), CST2 the second most (35), and just one genus is indicative for CST3. 


Let's see who they are
```{r}
length(indic_G$pval[indic_G$pval<0.05])

indic_sign <- as.data.frame(cbind(indic_G$pval[indic_G$pval<0.05],as.data.frame(indic_G$maxcls[indic_G$pval<0.05]),as.data.frame(indic_G$indcls[indic_G$pval<0.05])))

indic_sign2 <- cbind(indic_sign, as(tax_table(ps1)[rownames(indic_sign), "Genus"], "matrix"))

colnames(indic_sign2)[which(colnames(indic_sign2) %in% c("indic_G$pval[indic_G$pval < 0.05]","indic_G$maxcls[indic_G$pval < 0.05]", "indic_G$indcls[indic_G$pval < 0.05]"))] <- c("p_value_FDR_corrected","CST","Indicator_value")

indic_sign2$CST <- as.factor(as.character(indic_sign2$CST))

indic_sign2
```


### The significant indicator genera for CST1 with an IV >0.5: 
```{r}
indic_sign2$Genus[indic_sign2$CST == "1" & indic_sign2$Indicator_value>0.5]
```

### The significant indicator genera for CST2 with an IV >0.5: 
```{r}
indic_sign2$Genus[indic_sign2$CST == "2" & indic_sign2$Indicator_value>0.5]
```

### The significant indicator genera for CST3 with an IV >0.5: 
```{r}
indic_sign2$Genus[indic_sign2$CST == "3" & indic_sign2$Indicator_value>0.5]
```





### Genus level plots by CST
#### Barplots of indicator genera within the different CSTs (visualize top 5 indicator genera of CST 1, top 5 of CST 2 and the single indicator genus of CST 3)
```{r}
#genus glom
ps1_gen <- tax_glom(ps1, taxrank="Genus")
#relative abundance
ps1_gen_rel <- transform_sample_counts(ps1_gen, function(x) x / sum(x))
```


```{r fig.height=2, fig.width=6}
top_indicators <- c("Psychrobacter", "Bacillus", "Clostridium_sensu_stricto_1", "Lactobacillus", "Rahnella", "Cutibacterium", "Lawsonella", "Corynebacterium_1", "Anaerococcus", "Micrococcus", "Staphylococcus")

genus_colors <- c("Anaerococcus"= "#9E0142", "Bacillus"  = "#E25149", "Corynebacterium_1"  = "#F99153", "Lawsonella" = "#FDCB7A", "Micrococcus" = "#F0EC92", "Cutibacterium" = "#C7E89E", "Psychrobacter" = "#8ED1A4", "Rahnella" = "#56B0AC", "Staphylococcus" = "#387FB8", "Clostridium_sensu_stricto_1" = "#5E4FA2", "Lactobacillus" = "#978cc7", "Other" = "#a6a6a6")

df_rel_topI <- psmelt(ps1_gen_rel)
levels(df_rel_topI$Genus) <- c(levels(df_rel_topI$Genus), "Other")

df_rel_topI$Genus[!(df_rel_topI$Genus %in% top_indicators)] <- "Other"

#reorder
df_rel_topI$Genus <- as.factor(df_rel_topI$Genus)
df_rel_topI$Genus<-droplevels(df_rel_topI$Genus)
df_rel_topI$Genus<-factor(df_rel_topI$Genus, levels=c("Other", "Psychrobacter", "Bacillus", "Clostridium_sensu_stricto_1", "Lactobacillus", "Rahnella", "Cutibacterium", "Lawsonella", "Corynebacterium_1", "Anaerococcus", "Micrococcus", "Staphylococcus"))

pg <- ggplot(data = df_rel_topI, aes(x=Sample, y=Abundance, fill=Genus)) +
  geom_bar(stat = "identity", width = 0.9) + 
  scale_fill_manual(values=genus_colors) + 
  facet_grid(~ CST, scales = "free", labeller = label_both, space = "free") +
  theme(legend.text = element_text(size=12), axis.title.x = element_blank(), axis.ticks.x=element_blank(), axis.text.x = element_text(angle = 90, size = 8), axis.text.y = element_text( size = 14), axis.title.y = element_text(size = 14), strip.background =element_rect(fill="white"), strip.text.x = element_text(size=14, face = "bold"), legend.position = "bottom") + ylab("Relative abundance") 
```

#### Add color bars for subject type, time point, and Staphylococcus absolute abundance
Subject type:  
```{r fig.height=0.5, fig.width=6}
df_rel_topI1 <- df_rel_topI[,c("Sample", "Subject_type", "CST")]
df_rel_topI_m <- melt(df_rel_topI1, id.vars = c("Sample", "CST")) 
names(df_rel_topI_m)<-c("Sample", "CST", "Sub_ty", "Subject_type")

p2 <- ggplot(df_rel_topI_m, aes(x=Sample, y=Sub_ty)) + geom_tile(aes(fill=Subject_type), width = 0.7) + theme(legend.position = "right", legend.text = element_text(size=12), axis.title.x = element_blank(), axis.ticks.x=element_blank(), axis.ticks.y=element_blank(), axis.text.y=element_blank(), axis.text.x = element_text(angle = 90)) + scale_fill_manual(values = c("#00AFBB","#FF9505","#FC4E07")) + labs(y="")  
```

Time point:  
```{r fig.height=0.5, fig.width=6}
df_rel_topI2 <- df_rel_topI[,c("Sample", "Time_point_within_day_long", "CST")]
df_rel_topI2_m <- melt(df_rel_topI2, id.vars = c("Sample", "CST")) 
names(df_rel_topI2_m)<-c("Sample", "CST", "Time_point_within_day_long", "Daily_timepoint")
df_rel_topI2_m$Daily_timepoint[df_rel_topI2_m$Daily_timepoint == "non_exposed"]<-"NA"

p3 <- ggplot(df_rel_topI2_m, aes(x=Sample, y=Time_point_within_day_long)) + geom_tile(aes(fill=Daily_timepoint), width = 0.7) + theme(legend.position = "right", legend.text = element_text(size=12), axis.title.x = element_blank(), axis.ticks.x=element_blank(), axis.ticks.y=element_blank(), axis.text.y=element_blank(), axis.text.x = element_text(angle = 90)) + scale_fill_manual(values = c("#b2e7ea","#008c95","#002325","white")) + labs(y="")  
```

Staphylococcus estimated absolute abundance:
```{r fig.height=1, fig.width=6}
df_rel4 <- df_rel_topI[,c("Sample", "Abundance", "Genus", "CST", "Quantity_per_swab", "Time_point_within_day_long", "Subject_type", "Day", "Subject_ID")]

df_rel4_Staph <- df_rel4[df_rel4$Genus == "Staphylococcus",]
df_rel4_Staph$Stap_abs_abund_swab <- df_rel4_Staph$Abundance * df_rel4_Staph$Quantity_per_swab

## Estimated absolute abundance:
p4 <- ggplot(df_rel4_Staph, aes(x=Sample, y=Stap_abs_abund_swab)) + geom_bar(stat = "identity", width = 0.85, aes(fill="Genus")) + scale_y_continuous(position = 'right') + theme(legend.position = "none", axis.title.x = element_blank(), axis.ticks.x=element_blank(),   axis.text.x = element_text(angle = 90), axis.title.y.right = element_text(angle=0, vjust=0.5, size=11)) + scale_fill_manual(values = c("#387FB8")) + labs(y="Estimated absolute abundance \nof Staphylococcus")   + scale_y_log10(position="right")
```


```{r, fig.height=6, fig.width=15}
p1.1 <- pg + theme(axis.text.x=element_blank(), legend.position = "right") + facet_grid(. ~ CST, scales = "free_x", labeller = label_both, space="free") + guides(fill=guide_legend(nrow=12)) +theme(axis.text.y = element_text( size = 8), axis.title.y = element_text(size = 11), strip.background =element_rect(fill="white"), strip.text.x = element_text(size=11, face = "bold"), legend.text= element_text(size=11), legend.title = element_text(size=11))

p2.1 <- p2 + facet_grid(. ~ CST, scales = "free_x", labeller = label_both, space="free") + theme(strip.background = element_blank(), strip.text.x = element_blank(), axis.text.x = element_blank(), legend.position = "right", legend.text= element_text(size=11),legend.title = element_text(size=11), axis.title.y = element_text(size = 8)) + guides(fill=guide_legend(nrow=1)) + labs(fill="Subject_type")

p3.1 <- p3 + facet_grid(. ~ CST, scales = "free_x", labeller = label_both, space="free") + theme(strip.background = element_blank(), strip.text.x = element_blank(), axis.text.x = element_blank(), legend.position = "right", legend.background = element_rect(fill = "transparent", colour = NA),
    legend.box.background = element_rect(fill = "transparent", colour = NA), legend.text= element_text(size=11), legend.title = element_text(size=11)) + guides(fill=guide_legend(nrow=2)) + labs(fill="Daily_timepoint")

p4.1 <- p4 + facet_grid(. ~ CST, scales = "free_x", labeller = label_both, space="free") + theme(strip.background = element_blank(), strip.text.x = element_blank(), axis.text.x = element_blank(), legend.position = "none") 

fig2 <- cowplot::plot_grid(p1.1, p4.1, p2.1, p3.1, ncol = 1, align = "v", axis = "lr", rel_heights = c(0.8, 0.2, 0.15, 0.15))

fig2
```


## **Staphylococcal bacterial load (qPCR)** 
```{r fig.height=5, fig.width=5}
plot_load_Staph1 <- ggplot(df_rel4_Staph[df_rel4_Staph$Subject_type != "non_exposed",], aes(x = Time_point_within_day_long, y = Stap_abs_abund_swab, fill = Subject_type, color=Subject_type)) + geom_boxplot(outlier.color="NA", alpha = 0.5) +  geom_dotplot(binaxis = "y", stackdir = "center", alpha = 0.8, position=position_dodge(0.75), dotsize = 0.5) + theme(axis.title.y = element_text(size=12, face = "bold"), axis.text.y = element_text(size=16), axis.text.x = element_text(size=12, face = "bold", angle = 0), axis.title.x = element_blank(), legend.text = element_text(size=10, face = "bold"), legend.title = element_blank(), panel.grid.major = element_blank(), panel.background = element_blank(), panel.border = element_blank(), axis.line = element_line(colour = "black"), title = element_text(size = 14, face = "bold"), legend.position = "bottom") + labs(y="Estimated absolute abundance \nof Staphylococcus per sample") + stat_boxplot(geom ='errorbar')   + scale_fill_manual(values = c("#00AFBB","#FC4E07")) + scale_color_manual(values = c("#00AFBB","#FC4E07")) + scale_y_log10() + scale_x_discrete(labels=c("A_morning" = "Morning", "B_after_first_unloading" = "1", "C_after_second_unloading" = "2"))

plot_load_Staph1
```


```{r eval=FALSE, include=FALSE}
#saveRDS(ps1_hell, "./R_DATA/ps1_hell_pig_truck_drivers.RData")
#saveRDS(ps1, "./R_DATA/ps1_pig_truck_drivers.RData")
```



